vendor/processors/__init__.py

```diff
@@ -1,4 +1,18 @@
-from vendor.config import VENDOR_PAYMENT_PROCESSOR
 from django.utils.module_loading import import_string
+from django.core.exceptions import ObjectDoesNotExist
 
-PaymentProcessor = import_string('vendor.processors.{}'.format(VENDOR_PAYMENT_PROCESSOR))
+from vendor.config import PaymentProcessorSiteConfig
+from vendor.processors.authorizenet import AuthorizeNetProcessor
+from vendor.processors.base import PaymentProcessorBase
+from vendor.processors.dummy import DummyProcessor
+from siteconfigs.models import SiteConfigModel
+
+
+def get_site_payment_processor(site):
+    site_processor = PaymentProcessorSiteConfig()
+    try:
+        return import_string(f"vendor.processors.{SiteConfigModel.objects.get(site=site, key=site_processor.key).value['payment_processor']}")
```

Comment:
you can probably use `PaymentProcesorSiteConfig.instance` here; instead of a try/except, check if instance and return the default if not

Reply:
e.g. 
`processor = PaymentProcesorSiteConfig.instance
if processor:
    return import_string(f"vendor.processors.{processor.value['payment_processor']}")
else:
    return import_string(f"vendor.processors.{site_processor.default['payment_processor']}")`

Reply:
i would also pass in the site to PaymentProcessorSiteConfig

Reply:
lol you are correct. Added the try/catch first and then I implemented the instance which handles that jajajaj Thanks for noticing.